Quran/AyahInfoPersistenceStorage.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  AyahInfoPersistenceStorage.swift
+//  Quran
+//
+//  Created by Ahmed El-Helw on 5/12/16.
+//  Copyright © 2016 Quran.com. All rights reserved.
+//
+
+import Foundation
+import SQLite
+
+struct AyahInfoPersistenceStorage {
```

Comment:
i understand that your intention was for this to adopt the `SQLitePersistenceStorage` protocol. however, some thoughts - it seems to me that the library i pulled for SQLite attempts to be more expressive, and while it gives the person the option to write sql, it doesn't require it (as you can see in the example below).

that having been said, i am wondering if you'd be okay with `AyahInfoPersistenceStorage` which just has methods like `getAyahInfoForAyah`, `getAyahInfoForPage`, and `getAyahInfoForXY` for example instead of actually having to map this to something used by `execute`.

moreover, 2 other questions:
1. who is responsible for calling this on a background thread? i saw you did some stuff with this for the various `DataRetriever` classes for example, but wondering about this.
2. if these methods are very low level (i.e. given some parameters, return a list of `AyahInfo` objects, for example, then where does "business logic" on top of these methods go (for example, an ayah may have multiple `AyahInfo` entries (one per line), so what if we want to combine them into one `AyahInfo`, where would that go?